addons/account_edi/models/account_edi_format.py

```diff
@@ -29,41 +29,91 @@ class AccountEdiFormat(models.Model):
     # Export method to override based on EDI Format
     ####################################################
 
-    def _export_invoice_to_attachment(self, invoice):
+    def _export_invoice_to_attachment_vals(self, invoice):
         """ Create the file content representing the invoice.
 
-        :param invoice: the invoice to encode.
-        :returns: a dictionary (values are compatible to create an ir.attachment)
-        * name : the name of the file
-        * datas : the content of the file,
-        * res_model : 'account.move',
-        * res_id: the id of invoice
-        * mimetype : the mimetype of the attachment
+        :param invoice: The invoice to encode.
+        :returns:       A dictionary (values are compatible to create an ir.attachment).
+        * name :        The name of the file.
+        * datas :       The content of the file.
+        * res_model :   'account.move'
+        * res_id:       The id of invoice.
+        * mimetype :    The mimetype of the attachment
         """
         # TO OVERRIDE
         self.ensure_one()
         return False
 
-    def _export_invoice_to_embed_to_pdf(self, pdf_content, invoice):
+    def _export_invoice_to_embed_to_pdf(self, pdf_content, edi_document):
```

Comment:
I understand the concern, but I think a hook returning the content allows for more flexibility. For example, we know that some EDIs require the pdf to be embedded in b64 in the xml, this would not be possible without this hook.

Reply:
Following a meeting, it was decided that having pdf_content in the hook was not necessary since it's not use as of now and the idea behind it was a weirdness in old Belgian EDI. Since it might never be used, we decided to go with the most simple solution.